oktapam/data_source_secrets_test.go

```diff
@@ -0,0 +1,143 @@
+package oktapam
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+)
+
+func TestAccDatasourceSecretList(t *testing.T) {
+	checkTeamApplicable(t, true)
+
+	resourceGroupName := fmt.Sprintf("test_acc_resource_group_%s", randSeq())
+	delegatedAdminGroupName := fmt.Sprintf("test_acc_resource_group_dga_%s", randSeq())
+	projectName := fmt.Sprintf("test_acc_resource_group_project_%s", randSeq())
+	initialTopLevelFolderName := fmt.Sprintf("test_acc_secret_folder_%s", randSeq())
+	topLevelFolderSecurityPolicyName := fmt.Sprintf("test_acc_secret_folder_security_policy_%s", randSeq())
+	secret1Name := fmt.Sprintf("test_acc_secret_%s", randSeq())
+	secret2Name := fmt.Sprintf("test_acc_secret_%s", randSeq())
+
+	initConfig := createTestAccDatasourceSecretsCreateConfig(delegatedAdminGroupName, resourceGroupName, projectName, initialTopLevelFolderName, topLevelFolderSecurityPolicyName, secret1Name, secret2Name)
+
+	prefix := "data.oktapam_secrets"
+	dataSource1Name := fmt.Sprintf("%s.%s", prefix, "data1")
+	data1Config := createTestAccDatasourceSecretsDataConfig("data1", fmt.Sprintf("/%s/%s", initialTopLevelFolderName, secret1Name))
+	dataSource2Name := fmt.Sprintf("%s.%s", prefix, "data2")
+	data2Config := createTestAccDatasourceSecretsDataConfig("data2", fmt.Sprintf("/%s", initialTopLevelFolderName))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccResourceGroupCheckDestroy(resourceGroupName),
+		Steps: []resource.TestStep{
+			{
+				Config: initConfig,
+			},
+			{
+				Config: fmt.Sprintf("%s\n%s", initConfig, data1Config),
+				Check:  resource.TestCheckResourceAttr(dataSource1Name, fmt.Sprintf("%s.#", attributes.Secrets), "1"),
+			},
+			{
+				Config: fmt.Sprintf("%s\n%s", initConfig, data2Config),
+				Check:  resource.TestCheckResourceAttr(dataSource2Name, fmt.Sprintf("%s.#", attributes.Secrets), "2"),
```

Comment:
can you add a comment here to make it clearer what we're testing? I think the first one is 1 because it's getting an individual secret and the second is 2 because it's getting all secrets in the parent folder?

Reply:
Done